gnd/src/main/java/com/google/android/gnd/system/auth/GoogleAuthenticationManager.java

```diff
@@ -53,6 +54,9 @@
   @Hot(replays = true)
   private final Subject<SignInState> signInState = BehaviorSubject.create();
 
+  @Inject
+  LocalValueStore localValueStore;
```

Comment:
`GoogleAuthenticationManager` is a low-level component with no app-specific dependencies. As its name suggests, it's scoped as an abstraction of authentication details. Since ToS isn't part of the actual low-level auth process, it wouldn't make sense that this class knows about / depends on them. 

Also, `signOut()` is the method invoke when we want to request a sign out. It doesn't get called if the user gets signed out for other reasons (token expiration, account blocked, etc.).  In that case it would be better to listen to an event that indicates sign out.

The cleanest way to achieve both of these goals would be to create a `TermsOfServiceManager` which would subscribe to `authManager.getSignInState()`. It could then just clear the terms status when the user signs out. Wdyt?

Reply:
fixed

Reply:
added TermsOfServiceManager